src/js/todoFunctionality.js

```diff
@@ -0,0 +1,116 @@
+import auth from './auth';
+
+const todos = [];
+
+const userIdCharacters =
+  '0123456789abcdefghijklmnopqrstuvwxyz!@#$%^&.*()ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+
+const toggleCompleted = (index, refreshTodo) => {
+  todos[index].completed = !todos[index].completed;
+  if (todos[index].completed) {
+    todos[index].completed_at = new Date().toString();
+  } else {
+    todos[index].completed_at = null;
+  }
+  refreshTodo();
+};
+
+const renderTodos = () => {
+  const todoList = document.getElementById('todoList');
+  todoList.innerHTML = '';
+  todos.forEach((todo, index) => {
+    const li = document.createElement('li');
+
+    const creationDate = new Date(todo.created_at).toLocaleString();
+
+    li.innerHTML = `
+          <li class="flex w-full cursor-pointer items-center justify-between rounded-t-lg border-b-2 bg-blue-0 p-4 outline-none">
+              <div class="flex max-w-[90%] items-center gap-4">
```

Comment:
Are you adding `li` inside `li`?

Let's have a function that would generate a single todo li, also note that `textContent` is safer than `innerHTML` for user generated content such as the todo task.